addons/mass_mailing/models/mass_mailing.py

```diff
@@ -133,6 +133,24 @@ def _compute_contact_nbr(self):
         for mailing_list in self:
             mailing_list.contact_nbr = data.get(mailing_list.id, 0)
 
+    @api.multi
+    def write(self, vals):
+        # Prevent archiving used mailing list
+        if 'active' in vals and not vals.get('active'):
+            mass_mailings = self.env['mail.mass_mailing'].search_read([
+                ('state', '!=', 'done'),
+                ('contact_list_ids', 'in', self.ids),
+            ], ['name'])
+
+            if len(mass_mailings) > 0:
+                raise UserError(
+                    _("Sorry, you can't archive the mailing list \"%s\" because it is used in the following mailing(s): %s.") % (
+                        self.name,
```

Comment:
This will crash if len(self) > 1. You simply do "At least one of the mailing list you are trying to archive is used in an ongoing mailing campaign."

No need for a fancy message.